.github/workflows/report-package-size.yml

```
@@ -0,0 +1,30 @@
+name: Report Package Size
+
+on: [pull_request]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out the application's source code
+        uses: actions/checkout@v2
+
+      - name: Install node
+        uses: actions/setup-node@v1
+        with:
+          node-version: 12
+          registry-url: https://npm.pkg.github.com
+          scope: "@university-of-york"
+          always-auth: true
+
+      - name: Check and report changes in the application's package size
+        uses: preactjs/compressed-size-action@e2bfc4b346d78f9a3c7e8a2e22bcbd285a25fafa
+        with:
+          repo-token: ${{ secrets.GITHUB_TOKEN }}
+          build-script: package
+          pattern: .serverless/*.zip
+          compression: none
+          minimum-change-threshold: 1024
```

Personally, I'd be interested in changes smaller than 1Mb - although I'm struggling to quantify exactly where my interest would start. Maybe 256kb? I want the deployment to be as small as it can reasonably be. 50Mb is the Lambda hard limit (I believe) and it'd be good to stay way below that (~16Mb at present). Having said that, I appreciate this is mainly for catching big jumps like what happened with Next.js v10, so I don't mind if we want to stick with 1Mb for now and see how we get on.

```suggestion
          minimum-change-threshold: 256
```

---

Just double-checked the documentation and [this value is in bytes](https://github.com/marketplace/actions/compressed-size-action#increasing-the-required-threshold) - so any change above 1KiB is flagged!

Thinking about it, this value is basically a threshold for "do I need to think about this PR more" - so upping it to 256 (i.e. 262144) might be sensible anyway.

---

I did not spot that the value was in bytes - thanks for checking. I don't know why I assumed it was in kb. In that case, I personally don't mind it being left how it is at 1kb - that might help us get a feeling for how our changes impact the bundle size more regularly than 256kb would.